### PR TITLE
Feature/core 1615 improve vade performance

### PR DESCRIPTION
**Pull request by wulfraem**
- [CORE-1615]

[CORE-1615]: https://evannetwork.atlassian.net/browse/CORE-1615

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,9 +4,31 @@ version = "0.0.1"
 authors = ["Philip Kaiser <philip.kaiser@evan.team>"]
```
Comment (Philser) [inline Cargo.toml]:
```suggestion
authors = [
    "Sebastian Wolfram <sebastian.wolfram@evan.team>",
    "Sebastian Dechant <sebastian.dechant@evan.team>",
    "Philip Kaiser <philip.kaiser@evan.team>",
]
```

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+        None => match fallback {
+            Some(value) => value,
+            None => {
+                panic!("no value for {} given", arg_name);
+            }
+        },
+    }
+}
+
+fn get_arg_arg(arg_name: &str) -> Result<Arg, Box<dyn std::error::Error>> {
```
Comment (Philser) [inline src/main.rs]: 🧐

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+        ))
+    } else if signer.starts_with("local") {
+        Box::new(LocalSigner::new())
+    } else {
+        panic!("invalid signer config")
+    }
+}
+
+fn get_vade(matches: &ArgMatches<'static>) -> Result<Vade, Box<dyn std::error::Error>> {
+    let target = get_arg(&matches, "target", Some("13.69.59.185"));
```
Comment (Philser) [inline src/main.rs]: Is this an actual target IP? Why is it a plaintext IP instead of a domain? WIll it never change?

**Review by Philser — APPROVED** [review]

File: src/main.rs (hunk 2 above)
Comment (wulfraem) [reply src/main.rs]: Renamed functions that deal with arguments here to have a clearer naming:

- `get_args` --> `get_argument_matches`
- `get_arg` --> `get_argument_value`
- `get_arg_arg` --> `get_clap_argument`

File: src/main.rs (hunk 3 above)
Comment (wulfraem) [reply src/main.rs]: Not for now but will be updated in future.

**Review by Philser — APPROVED** [review]

**Pull request merged**